modules/apps/29-fee/ica_test.go

```diff
@@ -0,0 +1,192 @@
+
+	packetID := channeltypes.NewPacketId(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 1)
+	packetFees, found := suite.chainA.GetSimApp().IBCFeeKeeper.GetFeesInEscrow(suite.chainA.GetContext(), packetID)
+	suite.Require().True(found)
+	suite.Require().Equal(expectedFee, packetFees.PacketFees[0].Fee)
+
+	interchainAccountAddr, found := suite.chainB.GetSimApp().ICAHostKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), ibctesting.FirstConnectionID, path.EndpointA.ChannelConfig.PortID)
+	suite.Require().True(found)
+
+	// fund the interchain account on chainB
+	coins := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100000)))
+	msgBankSend := &banktypes.MsgSend{
+		FromAddress: suite.chainB.SenderAccount.GetAddress().String(),
+		ToAddress:   interchainAccountAddr,
+		Amount:      coins,
+	}
+
+	res, err = suite.chainB.SendMsgs(msgBankSend)
+	suite.Require().NotEmpty(res)
+	suite.Require().NoError(err)
+
+	// prepare a simple stakingtypes.MsgDelegate to be used as the interchain account msg executed on chainB
+	validatorAddr := (sdk.ValAddress)(suite.chainB.Vals.Validators[0].Address)
+	msgDelegate := &stakingtypes.MsgDelegate{
+		DelegatorAddress: interchainAccountAddr,
+		ValidatorAddress: validatorAddr.String(),
+		Amount:           sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(5000)),
+	}
+
+	data, err := icatypes.SerializeCosmosTx(suite.chainA.GetSimApp().AppCodec(), []sdk.Msg{msgDelegate})
+	suite.Require().NoError(err)
+
+	icaPacketData := icatypes.InterchainAccountPacketData{
+		Type: icatypes.EXECUTE_TX,
+		Data: data,
+	}
+
+	// ensure chainB is allowed to execute stakingtypes.MsgDelegate
+	params := icahosttypes.NewParams(true, []string{sdk.MsgTypeURL(msgDelegate)})
+	suite.chainB.GetSimApp().ICAHostKeeper.SetParams(suite.chainB.GetContext(), params)
+
+	// build the interchain accounts packet
+	packet := buildInterchainAccountsPacket(path, icaPacketData.GetBytes(), 1)
+
+	// write packet commitment to state on chainA and commit state
+	commitment := channeltypes.CommitPacket(suite.chainA.GetSimApp().AppCodec(), packet)
+	suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.SetPacketCommitment(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 1, commitment)
+	suite.chainA.NextBlock()
+
+	err = path.RelayPacket(packet)
+	suite.Require().NoError(err)
+
+	// ensure escrowed fees are cleaned up
+	packetFees, found = suite.chainA.GetSimApp().IBCFeeKeeper.GetFeesInEscrow(suite.chainA.GetContext(), packetID)
+	suite.Require().False(found)
+	suite.Require().Empty(packetFees)
+
+	// assert the value of the account balance after fee distribution
+	postDistBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+	suite.Require().Equal(preEscrowBalance, postDistBalance)
```

Comment:
Hmmm, I am not sure I understand how it's possible that `postDistBalance` matches the `preEscrowBalance`... The former should have the recv and ack fees deducted but the timeout fee refunded and the latter should have all three (ack, recv, timeout) fees deducted, right?